Neos.ContentRepository.Core/Classes/NodeType/NodeType.php

```diff
@@ -38,6 +37,9 @@ final class NodeType
      */
     public readonly NodeTypeName $name;
 
+    /** @phpstan-ignore-next-line */
+    public readonly TetheredNodeTypeDefinitions $tetheredNodeTypeDefinitions;
+
```

Comment:
i applied a little hack here which is allowed even in php 8.3 and also documented. Its temporary but ugly.

All done to be forwards compatible with https://github.com/neos/neos-development-collection/pull/4999, i just backported the `TetheredNodeTypeDefinitions` vo and `NodeType::tetheredNodeTypeDefinitions`

cc @bwaidelich

Reply:
see also https://github.com/neos/neos-development-collection/issues/4523 where i explain the hack and https://www.php.net/manual/en/language.oop5.properties.php

> Readonly properties cannot be [unset()](https://www.php.net/manual/en/function.unset.php) once they are initialized. However, it is possible to unset a readonly property prior to initialization, from the scope where the property has been declared.

Reply:
we (christian basti me) discussed that we want to go with this hack for now with the vision that the nodeType will be at one point a simple model like:


```php
public function __construct(
  // ...
  public readonly TetheredNodeTypeDefinitions $tetheredNodeTypeDefinitions,
  // ...
) {}
```

Reply:
As stated in https://github.com/neos/neos-development-collection/issues/4228#issuecomment-2127964634 we will probably NOT have that "planned"/ beloved followup in Neos 9 as this is a bigger one and requires lots of work.

The assumption was that the followup will go in and we can remove the `__get` hack again. We said for the (at that time) less likely event that the followup doesnt come, we will revert this and fallback to regular getters.

But there is one _elephant_ in the room i was not aware of: Php8.4. It will come with support for such computed properties (see https://stitcher.io/blog/new-in-php-84#property-hooks-rfc) meaning we would have what we want at real language level supported.

The question is, can we live with this hack for the time until php 8.4 when we can refactor it to use the real `get` hook?
My worst fear, that php wouldnt support our syntax forever is gone now at least :D

Wdyt cc @kitsunet